src/vs/code/node/cli.ts

```diff
@@ -319,23 +319,89 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (platform() !== 'darwin') {
+			child = spawn(process.execPath, argv.slice(2), options);
 
-		if (args.wait && waitMarkerFilePath) {
-			return new Promise<void>(resolve => {
+			if (args.wait && waitMarkerFilePath) {
+				return new Promise<void>(resolve => {
 
-				// Complete when process exits
-				child.once('exit', () => resolve(undefined));
+					// Complete when process exits
+					child.once('exit', () => resolve(undefined));
 
-				// Complete when wait marker file is deleted
-				whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
-			}).then(() => {
+					// Complete when wait marker file is deleted
+					whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
+				}).then(() => {
 
-				// Make sure to delete the tmp stdin file if we have any
-				if (stdinFilePath) {
-					unlinkSync(stdinFilePath);
+					// Make sure to delete the tmp stdin file if we have any
+					if (stdinFilePath) {
+						unlinkSync(stdinFilePath);
+					}
+				});
+			}
+		} else {
+			const requiresWait = args['status'] || args['verbose'] || hasReadStdinArg;
+			const envVars: string[] = [];
+			for (const envKey in env) {
+				if (envKey === '_') {
+					continue;
 				}
-			});
+				const value = env[envKey];
+				envVars.push('--env');
+				envVars.push(envKey + '=' + value ?? '');
+			}
+			const openArgs: string[] = ['-n'];
+
+			let tmpfile = '';
+			let tmpfileError = '';
+			if (requiresWait) {
+				const time = new Date().getTime();
+				tmpfile = `${tmpdir()}/vscode-wait-transfer-${time}.log`;
```

Comment:
Maybe rather reuse what we already do above in:

https://github.com/microsoft/vscode/blob/77905c850e170eab00cac5ca190c7b1fe5ad43ba/src/vs/code/node/cli.ts#L217-L217

Could be a function too that is reused to create a random file.

Reply:
Fixed -- created a helper function